src/client.ts

```diff
@@ -4292,13 +4292,10 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             signPromise = this.http.requestOtherUrl<IThirdPartySigned>(Method.Post, url);
         }
 
-        let queryParams: QueryDict = {};
+        const queryParams: QueryDict = {};
         if (opts.viaServers) {
             queryParams.server_name = opts.viaServers;
```

Comment:
A comment saying this is deprecated and giving a suggestion on when it could be removed feels like it's probably appropriate here?

Reply:
I've added a comment about the deprecation. I'm not sure what recommendation to give on when to remove `server_name` though. I guess unless the SDK's minimum server version is increased beyond the (future) spec version that actually removes `server_name`, it cannot be removed at all?

Reply:
Actually ... that was garbage. `server_name` can be dropped once the minimum version is raised to at least the version that introduced `via` (>1.11). Have updated the comment.